### ISSUE TITLE
Produced ps / pdf files are broken on Mac OS 10.14

### ISSUE THREAD

**Issue by jbmuir** [GenericMappingTools/GMT.jl#352]
PDF / PS files produced by, e.g. 
```
using GMT
plot(1:10, rand(10), lw=1, lc=:blue, marker=:square,
     markeredgecolor=0, size=0.2, markerfacecolor=:red, title="Hello World",
     xlabel="Spoons", ylabel="Forks", fmt=:pdf, savefig="test.pdf", show=true)
```
cannot be opened in Preview - interestingly the preview in Finder (or the popup when pressing space) works.

**Comment by joa-quim** [comment]
Maybe you want to post this in the GMT Forum where the Mac users are. This is almost for sure a Preview issue. Note that at the end it's always GMT who does the file creation. If you add the **Vd=1** option to the above plot command, you can see which commands is GMT really executing.

Also, by specifying the file name, ``savefig="test.pdf"``, there is no further need to use the ``fmt=:pdf`` because the format is fetch from file extension.

**Comment by jbmuir** [comment]
The file also will not open in adobe acrobat or affinity designer so I don't think its a preview problem

**Comment by joa-quim** [comment]
As I said, it would be better to have this discussion in the [Forum ](https://forum.generic-mapping-tools.org). I don't understand what the problem might be. The example is just a very simple GMT plot where a PS file is converted to PDF. Does the PS file opens fine for you?

**Comment by jbmuir** [comment]
The temporary file created during processing opens, but the final cropped version saved in the working directory does not.

**Comment by joa-quim** [comment]
By cropped you mean the PDF version?
What happens if you do the conversions manually? e.g.
```
gmt psconvert GMTjl_tmp.ps -A -Tf
```
or
```
gmt psconvert GMTjl_tmp.ps -A -Tg
```
and then open the respective converted files.

**Comment by jbmuir** [comment]
They open fine. By cropped I meant the final files that are saved in the working directory that have the rest of the page removed

**Comment by joa-quim** [comment]
OK, so this seems to be pointing to the problem being with the call to psconvert done in the ``showfig`` julia function. Still don't get what could go wrong in Mac (or the implicit call to ghostscript). Is the problem only with PDFs or happens with other formats as well?

**Comment by jbmuir** [comment]
EPS, TIF, JPG and PNG all work, only PDF and PS do not

**Comment by joa-quim** [comment]
Hmm, can't make any sense out of it. It's the same command that converts to each of those formats. But the conversion to PDF fails, though it works in command line!

You say it works for EPS but fail for PS. Something wrong here because PS is what we start with. Can you attach here the original PS and converted one (zipped)?

**Comment by jbmuir** [comment]
[PS_test.zip](https://github.com/GenericMappingTools/GMT.jl/files/4333263/PS_test.zip)
Sure - I agree that it is really strange; sorry for pestering you!

**Comment by joa-quim** [comment]
No problem. It's part of the deal trying to fix things. BTW, there might be a problem with wrapper because your ``.ps`` file is actually a PNG one. Just renamed it to ``.png`` and it opened fine. 
Can you please send the ``.pdf`` too?

**Comment by jbmuir** [comment]
I've attached the output; it is also a png though
[test.pdf](https://github.com/GenericMappingTools/GMT.jl/files/4333300/test.pdf)

**Comment by joa-quim** [comment]
So, for some reason in your case the conversion to both pdf or ps turn out to create a png but with a wrong extension!!!!!

Hard for me to debug this. I would need to debug the ``fname_out`` function in ``common_options.jl`` to find out why it screws in your case.

**Comment by htyeim** [comment]
> PDF / PS files produced by, e.g.
> 
> ```
> using GMT
> plot(1:10, rand(10), lw=1, lc=:blue, marker=:square,
>      markeredgecolor=0, size=0.2, markerfacecolor=:red, title="Hello World",
>      xlabel="Spoons", ylabel="Forks", fmt=:pdf, savefig="test.pdf", show=true)
> ```
> 
> cannot be opened in Preview - interestingly the preview in Finder (or the popup when pressing space) works.

Did you run these code in a Jupyter notebook/jupyterlab? I had the same problem when I outputted figures in jupyterlab. You may try to paste the code into a jl file then run it directly.

**Comment by jbmuir** [comment]
Yes I did run it in Jupyter, and can confirm that it works in the Julia REPL. I guess it needs to generate a png / svg or something to render in the notebook and that overrides the pdf creation?

**Comment by joa-quim** [comment]
OK, thanks that's new info. When running from Jupiter I indeed create PNGs for automatic visualization. Have to look back and revisit file renaming policy.

**Comment by joa-quim** [comment]
This issue has been fixed by #383
